drivers/soundwire/cadence_master.c

```diff
@@ -891,7 +891,10 @@ irqreturn_t sdw_cdns_irq(int irq, void *dev_id)
 			     CDNS_MCP_INT_SLAVE_MASK, 0);
 
 		int_status &= ~CDNS_MCP_INT_SLAVE_MASK;
-		schedule_work(&cdns->work);
+
+		/* if the interrupt disable is in process, don't schedule a workqueue */
+		if (cdns->interrupt_enable)
```

Comment:
@plbossart the comment and the logic are contradicting? Did you mean !cdns->interrupt_enable?

Reply:
yes, thanks for spotting this.